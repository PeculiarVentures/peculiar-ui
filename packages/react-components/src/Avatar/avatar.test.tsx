import React from 'react';
import {
  renderWithWrapper as render, screen, fireEvent,
} from '../test-utils';
import { Avatar } from './index';

const src = 'https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60';

describe('<Avatar />', () => {
  describe('Avatar render variants', () => {
    it('should render as default', () => {
      const {
        asFragment,
      } = render(<Avatar />);

      const img = screen.getByRole('img');

      expect(img).toBeInTheDocument();
      expect(img.tagName).toBe('svg');
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render as image', () => {
      const {
        asFragment,
      } = render(<Avatar src={src} />);

      const img = screen.getByRole('img');

      expect(img).toBeInTheDocument();
      expect(img.tagName).toBe('IMG');
      expect(img.getAttribute('src')).toEqual(src);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render 1 initial with a 1-word name', () => {
      const {
        asFragment,
      } = render(<Avatar name="First" />);

      expect(screen.getByText('F')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render 2 initials with a 2-word name', () => {
      render(<Avatar name="First Last" />);
      expect(screen.getByText('FL')).toBeInTheDocument();
    });

    it('should render 2 initials with a 3-word name', () => {
      render(<Avatar name="First Middle Last" />);
      expect(screen.getByText('FM')).toBeInTheDocument();
    });

    it('should render initials with fn', () => {
      const {
        asFragment,
      } = render(<Avatar name="First" getInitials={(str) => str.split('')[0]} />);

      expect(screen.getByText('F')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('should prioritize render as image over initials', () => {
      const {
        asFragment,
      } = render(<Avatar src={src} name="First" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have test id', () => {
      const {
        asFragment,
      } = render(<Avatar data-testid="test-id" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have class name', () => {
      const {
        asFragment,
      } = render(<Avatar className="test-cls" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('shoul change initials render element', () => {
      const {
        asFragment,
      } = render(
        <Avatar
          name="First"
          renderInitials={(props) => (
            // eslint-disable-next-line jsx-a11y/heading-has-content
            <h1 {...props} />
          )}
        />,
      );

      expect(screen.getByText('F')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Avatar render sizes', () => {
    const sizes: React.ComponentProps<typeof Avatar>['size'][] = [
      'small',
      'medium',
      'large',
    ];

    sizes.forEach((size) => {
      it(`size "${size}"`, () => {
        const {
          asFragment,
        } = render(
          <Avatar size={size} />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('Avatar handlers', () => {
    it('should call onError handler', () => {
      const handleErr = jest.fn();

      render(<Avatar src="broken.png" onError={handleErr} />);

      const img = screen.getByRole('img');

      fireEvent.error(img);

      expect(handleErr).toHaveBeenCalledTimes(1);
    });

    it('should call onLoad handler', () => {
      const handleLoad = jest.fn();

      render(<Avatar src={src} onLoad={handleLoad} />);

      const img = screen.getByRole('img');

      fireEvent.load(img);

      expect(handleLoad).toHaveBeenCalledTimes(1);
    });
  });
});
