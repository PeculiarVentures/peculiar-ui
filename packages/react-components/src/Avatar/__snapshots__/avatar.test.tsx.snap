// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Avatar render variants should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-gray-7);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  height: var(--pv-size-base-7);
  width: var(--pv-size-base-7);
}

<div
    class="my-class-name emotion-0"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 10 13"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.998 3.174c0 1.594 1 2.674 2 2.674s2-1.08 2-2.674c0-1.595-1-2.674-2-2.674s-2 1.08-2 2.674zM5 8.858c-1.249 0-2.383-.448-3.308-1.146C.95 8.064.5 8.878.5 9.812v2.12h9V9.81c0-.918-.46-1.746-1.193-2.098C7.383 8.41 6.25 8.858 5 8.858z"
        fill="#fff"
        stroke="#fff"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Avatar /> Avatar render variants should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-gray-7);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  height: var(--pv-size-base-7);
  width: var(--pv-size-base-7);
}

<div
    class="emotion-0"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 10 13"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.998 3.174c0 1.594 1 2.674 2 2.674s2-1.08 2-2.674c0-1.595-1-2.674-2-2.674s-2 1.08-2 2.674zM5 8.858c-1.249 0-2.383-.448-3.308-1.146C.95 8.064.5 8.878.5 9.812v2.12h9V9.81c0-.918-.46-1.746-1.193-2.098C7.383 8.41 6.25 8.858 5 8.858z"
        fill="#fff"
        stroke="#fff"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Avatar /> sizes size "large" 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-gray-7);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  height: var(--pv-size-base-8);
  width: var(--pv-size-base-8);
}

<div
    class="emotion-0"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 10 13"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.998 3.174c0 1.594 1 2.674 2 2.674s2-1.08 2-2.674c0-1.595-1-2.674-2-2.674s-2 1.08-2 2.674zM5 8.858c-1.249 0-2.383-.448-3.308-1.146C.95 8.064.5 8.878.5 9.812v2.12h9V9.81c0-.918-.46-1.746-1.193-2.098C7.383 8.41 6.25 8.858 5 8.858z"
        fill="#fff"
        stroke="#fff"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Avatar /> sizes size "medium" 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-gray-7);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  height: var(--pv-size-base-7);
  width: var(--pv-size-base-7);
}

<div
    class="emotion-0"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 10 13"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.998 3.174c0 1.594 1 2.674 2 2.674s2-1.08 2-2.674c0-1.595-1-2.674-2-2.674s-2 1.08-2 2.674zM5 8.858c-1.249 0-2.383-.448-3.308-1.146C.95 8.064.5 8.878.5 9.812v2.12h9V9.81c0-.918-.46-1.746-1.193-2.098C7.383 8.41 6.25 8.858 5 8.858z"
        fill="#fff"
        stroke="#fff"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Avatar /> sizes size "small" 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: var(--pv-color-gray-7);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  height: var(--pv-size-base-6);
  width: var(--pv-size-base-6);
}

<div
    class="emotion-0"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 10 13"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.998 3.174c0 1.594 1 2.674 2 2.674s2-1.08 2-2.674c0-1.595-1-2.674-2-2.674s-2 1.08-2 2.674zM5 8.858c-1.249 0-2.383-.448-3.308-1.146C.95 8.064.5 8.878.5 9.812v2.12h9V9.81c0-.918-.46-1.746-1.193-2.098C7.383 8.41 6.25 8.858 5 8.858z"
        fill="#fff"
        stroke="#fff"
      />
    </svg>
  </div>
</DocumentFragment>
`;
